_integration-test/run.py

```diff
@@ -0,0 +1,176 @@
+import subprocess
+import os
+from functools import lru_cache
+import requests
+import pytest
+import sentry_sdk
+import time
+import json
+
+SENTRY_CONFIG_PY = "sentry/sentry.conf.py"
+SENTRY_TEST_HOST = os.getenv("SENTRY_TEST_HOST", "http://localhost:9000")
+TEST_USER = "test@example.com"
+TEST_PASS = "test123TEST"
+TIMEOUT_SECONDS = 60
+
+
+def run_shell_command(cmd, **kwargs) -> subprocess.CompletedProcess:
+    return subprocess.run(cmd, shell=True, check=True, **kwargs)
+
+
+def poll_for_response(request: str, session: requests.Session) -> requests.Response:
+    for i in range(TIMEOUT_SECONDS):
+        response = session.get(request, headers={"Referer": SENTRY_TEST_HOST})
+        if response.status_code == 200:
+            break
+        time.sleep(1)
+    assert response.status_code == 200
+    return response
+
+
+@lru_cache
+def get_sentry_dsn(session: requests.Session) -> str:
+    for i in range(TIMEOUT_SECONDS):
+        response = session.get(
+            f"{SENTRY_TEST_HOST}/api/0/projects/sentry/internal/keys/",
+            headers={"Referer": SENTRY_TEST_HOST},
+        )
+        if response.status_code == 200:
+            sentry_dsn = json.loads(response.text)[0]["dsn"]["public"]
+            if len(sentry_dsn) > 0:
+                break
+        time.sleep(1)
+    sentry_dsn = json.loads(response.text)[0]["dsn"]["public"]
+    assert len(sentry_dsn) > 0
+    return sentry_dsn
+
+
+@pytest.fixture(scope="session", autouse=True)
+def configure_self_hosted_environment():
+    response = None
+    run_shell_command("docker compose --ansi never up -d")
+    for i in range(TIMEOUT_SECONDS):
+        try:
+            response = requests.get(SENTRY_TEST_HOST)
+        except requests.ConnectionError:
+            time.sleep(1)
+        if response is not None and response.status_code == 200:
+            break
+    assert response.status_code == 200
+
+    # Create test user
+    run_shell_command(
+        f"echo y | docker compose exec web sentry createuser --force-update --superuser --email {TEST_USER} --password {TEST_PASS}"
+    )
+
+
+@pytest.fixture(scope="function")
```

Comment:
`scope="function"` is the default
```suggestion
@pytest.fixture
```